app/server/utils/metrics/metrics.py

```diff
@@ -129,14 +129,19 @@ def calculate_metric(metric):
         db.session.close()
         return metric.metric_name, result
 
-
-    futures = []
-    for metric in metrics_list:
-        futures.append(executor.submit(calculate_metric, metric))
+    # After request, do things synchronously since time doesn't matter much
     data = {}
-    for future in futures:
-        metric_name, result = future.result()
-        data[metric_name] = result
+    if g.get('is_after_request'):
+        for metric in metrics_list:
+            metric_name, result = calculate_metric(metric)
+            data[metric_name] = result
+    else:
+        futures = []
```

Comment:
This is the one spot we still want flask-executor. It's supremely janky after request, but quite stable in the right context. It really really helps with performance here.